### PR TITLE
Marshmallow integration and more endpoints

### PR DESCRIPTION
**Pull request by finloop**
TODO:
- [x] Add sqlalchemy validators
- [x] Move models to separate files 
- [x] Write schemas for all models 
- [x] Begin marshmallow integration
- [x] Finish marshmallow integration
- [x] Rewrite all tests
- [x] Rewrite docs

### CODE REVIEW

File: src/extras/json.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import datetime
+
+from flask.json import JSONEncoder
+
+
+class CustomJSONEncoder(JSONEncoder):
+    """Add support for serializing timedeltas."""
+
+    def default(self, o):
```
Comment (pgronkievitz) [inline src/extras/json.py:9]: that's not documented

File: src/extras/parsers.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from flask import abort
+from flask import jsonify
+from flask import make_response
+from webargs.flaskparser import FlaskParser
+
+
+def set_attr_from_dict(object, data: dict):
+    """Set all non None object attrubutes from data."""
```
Comment (pgronkievitz) [inline src/extras/parsers.py:8]: arguments docs

File: src/extras/parsers.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    for key in data.keys():
+        if data[key] is not None:
+            setattr(object, key, data[key])
+
+
+def fail(msg, status=422):
+    """
+    Generates JSON error msg
```
Comment (pgronkievitz) [inline src/extras/parsers.py:16]: Wrong docstyle

File: src/models/server.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        Unique id
+    hostname : str
+        Hostname of server
+    cores : int
+        Number of cpu cores reserved
+    ram : int
+        MB of RAM memory reserved
+    gpus : int
+        Number of gpus reserved
+
```
Comment (pgronkievitz) [inline src/models/server.py:23]: unnecessary blank

File: src/models/user.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+from src.extensions import BCRYPT
+from src.extensions import DB
+
+
+class User(DB.Model):
+    """
+    Basic user database model.
+
+    ...
```
Comment (pgronkievitz) [inline src/models/user.py:12]: remove those dots

File: src/models/user.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    Attributes
+    ----------
+    id : int
+        Unique user identificator.
+    email : str
+        Email address.
+    password : str
+        Hashed password.
+
+    # TODO: Add docstring for methods
```
Comment (pgronkievitz) [inline src/models/user.py:23]: already documented

File: src/reserv/routes/api/auth/auth.py [hunk 7]
```diff
@@ -79,67 +73,32 @@ def login():
...
-        access_token
-
-    Examples
-    --------
-    .. code-block:: bash
-        $ curl -X POST -H 'token: bearer YOURTOKEN' https://example.com/api/auth/refresh
-    """
-    curent_user = get_jwt_identity()
-    return jsonify(access_token=create_access_token(identity=curent_user, fresh=False))
+        return jsonify(error="Invalid email or password."), 404
```
Comment (pgronkievitz) [inline src/reserv/routes/api/auth/auth.py:97]: wrong indentation

File: src/reserv/routes/api/servers/servers.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+from src.serializers.server import servers_schema
+from src.serializers.server import ServerSchema
+
+
+SERVERS = Blueprint("servers", __name__, url_prefix="/api/servers")
+
+# TODO: Replace parser with our own implementation
+
+
+@SERVERS.route("/hello", methods=["GET"])
```
Comment (pgronkievitz) [inline src/reserv/routes/api/servers/servers.py:23]: remove this placeholder

File: src/reserv/routes/api/servers/servers.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+def hello():
+    """Test api response."""
+    return jsonify(message="Hello from server API")
+
+
+@SERVERS.route("/all", methods=["GET"])
```
Comment (pgronkievitz) [inline src/reserv/routes/api/servers/servers.py:29]: should be `/`

File: src/reserv/routes/api/auth/auth.py (hunk 7 above)
Comment (finloop) [reply src/reserv/routes/api/auth/auth.py:97]: What's wrong with it? It seems fine to me.

Comment (pgronkievitz) [reply src/reserv/routes/api/auth/auth.py:97]: deleted code mislead me

**Pull request merged**